src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -1832,18 +1836,34 @@ impl Coordinator {
         let view_id = self.allocate_transient_id()?;
         let index_id = self.allocate_transient_id()?;
 
-        let cluster = self.catalog().active_cluster(session)?;
-        let target_replica_name = session.vars().cluster_replica();
-        let mut target_replica = target_replica_name
-            .map(|name| {
-                cluster.replica_id_by_name.get(name).copied().ok_or(
-                    AdapterError::UnknownClusterReplica {
-                        cluster_name: cluster.name.clone(),
-                        replica_name: name.to_string(),
-                    },
-                )
-            })
-            .transpose()?;
+        // If our query only depends on system tables, a LaunchDarkly flag is enabled, and a
+        // session var is set, then we automatically run the query on the mz_introspection cluster
+        let (cluster, mut target_replica) = match introspection::auto_run_on_introspection(
+            self.catalog(),
+            session,
+            source.depends_on(),
+        ) {
+            Some(cluster) => (cluster, None),
```

Comment:
I think the concern about target replicas above is that they won't be respected if they exist, which it still doesn't do. I think what you need to do is only run `auto_run_on_introspection` if `target_replica_name` is `None`.

Reply:
That makes sense, I was trying to not be overly restrictive, but just returning `None` if there is a target replica set is the cleaner approach. Updating now!